cmd/coordinated/metrics.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2015-2017 Diffeo, Inc.
+// This software is released under an MIT/X11 open source license.
+
+package main
+
+import (
+	"github.com/diffeo/go-coordinate/coordinate"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var coordinateSummary = prometheus.NewGaugeVec(
+	prometheus.GaugeOpts{
+		Namespace: "diffeo",
+		Subsystem: "coordinate",
+		Name:      "coordinate_summary",
+		Help:      "Summary of coordinate work specs",
+	},
+	[]string{
+		"namespace",
+		"work_spec",
+		"status",
+	},
+)
+
+func init() {
+	prometheus.MustRegister(coordinateSummary)
+}
+
+func observe(coord coordinate.Coordinate) {
+	for {
```

Comment:
You can't run this in a "hard" loop like this with no delays.  At the very very least put a `time.Sleep()` call.

(On the memory backend this will try to hold the single global lock and cause big problems.  On the postgres backend, on our busy systems this call is probably 20-30 seconds of database I/O.)

Ideally this function would take a `context.Context` with a cancellation channel and `select{}` on that and a timer.